Dockerfile

```diff
@@ -0,0 +1,72 @@
+# Based on OpenAI's mujoco-py Dockerfile
+
+FROM nvidia/cuda:10.0-runtime-ubuntu18.04 AS base
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN echo ttf-mscorefonts-installer msttcorefonts/accepted-mscorefonts-eula select true | debconf-set-selections \
+    && apt-get update -q \
+    && apt-get install -y --no-install-recommends \
+    build-essential \
+    curl \
+    ffmpeg \
+    git \
+    libgl1-mesa-dev \
+    libgl1-mesa-glx \
+    libglew-dev \
+    libosmesa6-dev \
+    net-tools \
+    parallel \
+    python3.7 \
+    python3.7-dev \
+    python3-pip \
+    rsync \
+    software-properties-common \
+    unzip \
+    vim \
+    virtualenv \
+    xpra \
+    xserver-xorg-dev \
+    ttf-mscorefonts-installer \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN curl -o /usr/local/bin/patchelf https://s3-us-west-2.amazonaws.com/openai-sci-artifacts/manual-builds/patchelf_0.9_amd64.elf \
+    && chmod +x /usr/local/bin/patchelf
+
+ENV LANG C.UTF-8
+
+RUN    mkdir -p /root/.mujoco \
+    && curl -o mjpro150.zip https://www.roboti.us/download/mjpro150_linux.zip \
+    && unzip mjpro150.zip -d /root/.mujoco \
+    && rm mjpro150.zip
+
+# Set the PATH to the venv before we create the venv, so it's visible in base.
+# This is since we may create the venv outside of Docker, e.g. in CI
+# or by binding it in for local development.
+ENV PATH="/imitation/venv/bin:$PATH"
+ENV LD_LIBRARY_PATH /usr/local/nvidia/lib64:/root/.mujoco/mjpro150/bin:${LD_LIBRARY_PATH}
+
+FROM base as python-req
+
+WORKDIR /imitation
+# Copy over just setup.py and __init__.py (including version)
+# to avoid rebuilding venv when requirements have not changed.
```

Comment:
Looks like we are always building a virtualenv at `venv` via `RUN ... ci/build_venv.sh`, so I'm not sure if this comment applies.

Reply:
Docker will cache previously built layers when the `Dockerfile` has not changed and all files are up to date

Reply:
Ah, gotcha, so the `RUN ... build_venv.sh` is only executed if the previous layers have changed.